languagetool-language-modules/pt/src/main/resources/org/languagetool/rules/pt/grammar.xml

```diff
@@ -20273,7 +20273,66 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA.
+            <rule default="temp_off">
+                <!-- This antipattern refers to constructions of the type 'the fact that...', for which
+                     a personal (i.e. inflected) infinitive can be used. -->
+                <antipattern>
+                    <token regexp="yes">fac?to|questão|possibilidade|hipótese|apesar</token>
+                    <token regexp="yes" skip="-1" postag_regexp="yes" postag="SPS00(:.+)?">
+                        d([eoa]|es[st][ea]|aquel[ea]|uma?|is[st]o)
+                        <exception postag_regexp="yes" postag="V.+"/>
+                    </token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>A questão de o segredo dever ter sido revelado não nos afeta.</example>
+                    <example>O fato do contrato já dever ter sido assinado é irrelevante.</example>
+                </antipattern>
+                <antipattern>
+                    <token postag="CS"/>
+                    <token postag="CS" skip="-1">
+                        de
+                        <exception postag_regexp="yes" postag="V.+"/>
+                    </token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>Apesar de ele dever já ter entrado, tranquei a porta.</example>
+                </antipattern>
+                <!-- A pesky little case, where 'a' + inf. actually generally fits..? -->
+                <antipattern>
+                    <token postag="SPS00">a</token>
+                    <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                    <example>A cidade ficou-lhe a dever não ter sido invadida.</example>
+                </antipattern>
+                <pattern>
+                    <token postag_regexp="yes" postag="SENT_START|[DNPARZ].+" min="1"/>
+                    <marker>
+                        <token postag_regexp="yes" postag="VMN.+">dever</token>
+                    </marker>
+                    <token postag_regexp="yes" postag="R." min="0"/>
+                    <token>ter</token>
+                    <token postag_regexp="yes" postag="V.P.+"/>
+                </pattern>
+                <message>O duplo infinitivo parece indicar um error de digitação. Se for o caso, substituta &quot;\2&quot; por <suggestion><match no="2" postag_regexp="yes" postag="VMN.+" postag_replace="VMIP3S0"/></suggestion>.</message>
```

Comment:
More suggestions (such as `deveria`) would likely increase apply rates. But this is also something we could check in the diff results. I'll leave it up to you if you want to leave it as it is.

Reply:
My initial assumption is that these instances of `dever` are typos for `deve`, not `deveria`. That's just a hunch, we'd need some real-world usage data to confirm, of course. I could totally add `deveria` to the suggestions, though, it's a quick thing to add.

Reply:
I think this is mostly the case, but in the era of “auto complete” features, I feel some people don't always type whole words all the time. We can see what kind of results we get and decide later.